components/WeatherForecastModal.js

```diff
@@ -0,0 +1,92 @@
+import {
+  WeatherForecastModalField,
+  ModalCloseButton,
+  Table,
+  DisplayedDaysContainer,
+  DisplayedDay,
+  Date,
+  Daytime,
+  WeatherDataContainer,
+  WeatherIcon,
+  Temperature,
+  Windspeed,
+} from "./styledComponents/WeatherForecasModal.styles";
+import WeekdayFromDateString from "./WeekdayFromDateString";
+import ArrowIcon from "./icons/ArrowWeatherIcon";
+
+export default function WeatherForecastModal({
+  filteredWeatherMorning,
+  filteredWeatherAfternoon,
+  filteredWeatherEvening,
+  onClickButton,
+}) {
+  // extracting the dates from the date-time-String with help of the static end value "06:00:00"
+  const dates = filteredWeatherMorning
+    .filter((date) => date.dt_txt.endsWith("06:00:00"))
+    .map((date) => date.dt_txt.split(" ")[0]);
+
+  return (
+    <>
+      <WeatherForecastModalField>
+        <ModalCloseButton onClick={onClickButton}>x</ModalCloseButton>
+        <Table>
+          <tbody>
+            <td>
+              {/*because of space issues and of the everchanging WeatherArray, the filteredWeatherArrays will only be mapped to the third day  */}
+              {filteredWeatherMorning.slice(0, 3).map((weekday, index) => (
+                <DisplayedDaysContainer key={index}>
```

Comment:
never use the index here. This is the source of many very weird bugs. The key must be tied to your data you a mapping over and be unique. you could take the daytime string.

Reply:
This is true for all instances where you used index. please double check your code and change it accordingly.